src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -156,41 +198,47 @@ public void start(Stage stage) {
 
           // iterate through all feature layers in the map
           for (Layer layer : mapView.getMap().getOperationalLayers()) {
-            final FeatureLayer featureLayer = (FeatureLayer) layer;
+            FeatureLayer featureLayer = (FeatureLayer) layer;
 
-            // identify any clicked feature
+            // start a process to identify any features at the clicked screen point
             ListenableFuture<IdentifyLayerResult> results = mapView.identifyLayerAsync(featureLayer, screenPoint, 1, false, 1);
             results.addDoneListener(() -> {
               try {
-                // get selected feature, if it is a point
+                // get the results of the identification process to determine whether a feature was clicked on
                 List<GeoElement> elementList = results.get().getElements();
+                // determine whether the clicked feature represents a point that can be moved
                 if (elementList.size() > 0 && elementList.get(0) instanceof ArcGISFeature && elementList.get(0).getGeometry().getGeometryType() == GeometryType.POINT) {
 
-                  // clicked on a feature, select it
+                  // clear the previous selection (to avoid selecting multiple features at the same time)
+                  featureLayer.clearSelection();
+
+                  // grab the new feature and make it selected
                   ArcGISFeature selectedFeature = (ArcGISFeature) elementList.get(0);
```

Comment:
rename to newSelectedFeature to avoid confusion with selectedFeature on line 235